### PR TITLE
Provision: localize provisioner plan and plan builder

### PR DESCRIPTION
**Pull request by dkropachev**
Based on: https://github.com/scylladb/scylla-cluster-tests/pull/4098

Purpose is to localize logic regarding provisioning and generalize it.
Preparation for https://trello.com/c/EyP2J8ig/1290-microsoft-azure-as-a-new-backend

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Tested - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-byo-longevity/545/

File: sdcm/provision/common/provision_plan.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                    instance_parameters=instance_parameters,
+                    count=node_count,
+                    tags=node_tags,
+                    names=node_names,
+            ):
+                LOGGER.info('%s: Instances has been provisioned using "%s":\n%s', self.name,
+                            provision_parameters.name, instances)
+                return instances
+            else:
+                LOGGER.error('%s: Failed has been provisioned using "%s"', self.name, provision_parameters.name)
```
Comment (vponomaryov) [inline sdcm/provision/common/provision_plan.py:33]: 'Failed has been provisioned using "%s"' sounds strange.
Maybe 'Failed to provision instances using "%s"'"?

File: sdcm/provision/common/provision_plan.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import logging
```
Comment (vponomaryov) [inline sdcm/provision/common/provision_plan.py:1]: Evgeniy would ask for the licence comment block.

File: sdcm/provision/common/provision_plan_builder.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
+import logging
```
Comment (vponomaryov) [inline sdcm/provision/common/provision_plan_builder.py:1]: Same here.

File: sdcm/provision/common/provision_plan.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/provision/common/provision_plan.py:33]: fixed

File: sdcm/provision/common/provision_plan_builder.py (hunk 3 above)
Comment (dkropachev) [reply sdcm/provision/common/provision_plan_builder.py:14]: added

File: sdcm/provision/common/provision_plan.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/provision/common/provision_plan.py:14]: added

**Review by vponomaryov — APPROVED** [review]

**Review by bentsi — APPROVED** [review]

**Pull request merged**